core/src/main/java/org/infinispan/security/impl/SubjectACL.java

```diff
@@ -38,4 +41,14 @@ public boolean matches(int permissionMask) {
    public String toString() {
       return "SubjectACL [roles=" + roles + ", mask=" + mask + "]";
    }
+
+   public EnumSet<AuthorizationPermission> getPermissions() {
+      EnumSet<AuthorizationPermission> permissions = EnumSet.noneOf(AuthorizationPermission.class);
+      for(AuthorizationPermission permission : AuthorizationPermission.values()) {
+         if ((mask & permission.getMask()) != 0) {
```

Comment:
@tristantarrant I tested the ReaderRole and there is something that seems weird.

Reply:
I created ```xml 
<role name="ReaderRole" permissions="READ"/>
```
```bash 
bin/cli.sh user create read -p "pass" -g ReaderRole
```

And I get in REST ALL permission in global, despite the fact user 'read' has only ReaderRole 
```json
"global":[
      "READ",
      "ALL",
      "ALL_READ"
   ],
```